src/Model_Hydro/MHD_Init_BField_ByFile.cpp

```diff
@@ -0,0 +1,369 @@
+#include "GAMER.h"
+
+#ifdef MHD
+#ifdef SUPPORT_HDF5
```

Comment:
Currently, the compilation will fail if we disable `SUPPORT_HDF5` since this function is still invoked in `Hydro_Init_ByFunction_AssignData()`. Given that one day, we may want to support loading vector potential in the format other than HDF5, we should probably just enclose all HDF5 calls in this function with `#ifdef SUPPORT_HDF5 ... #endif` and remove the `#endif // #ifdef SUPPORT_HDF5` at the end of this file.

But I'm open to other options.